nmsg/kafkaio.c

```diff
@@ -489,13 +512,52 @@ _kafka_delivery_cb(rd_kafka_t *rk, const rd_kafka_message_t *rkmessage, void *op
 		return;
 	if (rkmessage->err != RD_KAFKA_RESP_ERR_NO_ERROR) {
 		_nmsg_dprintf(2, "%s: got Kafka error %d: %s\n", __func__, rkmessage->err,
-			      rd_kafka_message_errstr(ctx->message));
+			      rd_kafka_err2str(rkmessage->err));
 		ctx->state = kafka_state_break;
 		rd_kafka_yield(rk);
 	}
+	ctx->error_retry = 0;
 	ctx->delivered++;
 }
 
+static void
+_kafka_rebalance_cb(rd_kafka_t *rk, rd_kafka_resp_err_t err, rd_kafka_topic_partition_list_t *partitions, void *opaque)
+{
+	rd_kafka_error_t *error = NULL;
+	rd_kafka_resp_err_t ret_err = RD_KAFKA_RESP_ERR_NO_ERROR;
```

Comment:
suggest s/ret_err/resp_err/

It seems gross that kafka has two types of "error".